src/System.Security.Cryptography.Algorithms/src/Internal/Cryptography/HashProviderDispenser.Unix.cs

```diff
@@ -130,7 +130,11 @@ public unsafe HmacHashProvider(IntPtr algorithmEvp, byte[] key)
 
                 fixed (byte* keyPtr = key)
                 {
-                    Check(Interop.libcrypto.HMAC_Init(out _hmacCtx, keyPtr, key.Length, algorithmEvp));
+                    _hmacCtx = Interop.Crypto.HmacCreate(keyPtr, key.Length, algorithmEvp);
+                    if (_hmacCtx == null)
```

Comment:
Great catch. Oops. Easy mistake to make when you just typed return nullptr on the other side. This would be a good Roslyn analyzer, safehandle returned from P/Invoke compared to null. ;)